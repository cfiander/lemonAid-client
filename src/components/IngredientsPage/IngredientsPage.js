import React, { Component } from 'react'
import RecipeContext from '../../contexts/RecipeContext'
import Ingredient from '../../components/IngredientItem/IngredientItem'
import { Link } from 'react-router-dom'
import Search from '../../components/Search/Search';

export default class IngredientsPage extends Component {
  static defaultProps = {
    match: {
      params: {}
    }
  }

  static contextType = RecipeContext


  render() {
    const { ingredientsList = [] } = this.context
    return (
      <div className="ingredients">
        <div className="searchForm">
          <Search />
        </div>
        <ul className="ingredientList">
          {ingredientsList.map((ingredient, i) =>
            <li key={ingredient.id} className="ingredient">
              <Ingredient
                key={i}
                id={ingredient.stateId}
                name={ingredient.name} />
            </li>
          )}
        </ul>
        <Link to={`/recipes`} className="basicButton">
          search recipes
        </Link>
      </div>
    )
  }
}
