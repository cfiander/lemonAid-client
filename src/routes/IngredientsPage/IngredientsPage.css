.shopping-list {
    width: 90%;
    margin: auto; 
    padding: 2.5%;
    color: whitesmoke;
    text-align: center;
    flex-wrap: wrap;
    display: flex;
  }
  
  .shopping-list > li {
    margin-bottom: 20px;
    padding: 20px;
  }
  
  .shopping-item {
    display: block;
    color: whitesmoke;
    font-style: italic;
    text-align: center;
    font-family: 'Playfair Display', serif;
    font-size: 1rem;
    margin-bottom: 15px;
    text-transform: uppercase;
  }