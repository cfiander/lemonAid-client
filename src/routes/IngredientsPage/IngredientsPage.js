import React, { Component } from 'react'
import RecipeContext from '../../contexts/RecipeContext'
import Ingredient from '../../components/IngredientItem/IngredientItem'
import './IngredientsPage.css'
import Search from '../../components/Search/Search';

export default class IngredientsPage extends Component {
    static defaultProps = {
        match: {
          params: {}
        }
    }

    static contextType = RecipeContext


    render() {
        const {ingredientsList = []} = this.context
        return (
        <section>
          <div className="searchForm">
          <Search/>
          </div>
          {/* The following commented out section is for a soon to come build with filtering optins */}
          {/* <ul className="filter-controls">
          <li>
            <select>
            <option value="" selected disabled hidden>Diet</option>
            <option value="keto">Keto</option>
            <option value="aleo">Paleo</option>
            <option value="rimal">Primal</option>
            <option value="frican">Pescetarian</option>
            <option value="egan">Vegan</option>
            <option value="egetarian">Vegetarian</option>
          </select>
          </li>
          <li>
          <select>
            <option value="" selected disabled hidden>Cuisine</option>
            <option value="african">African</option>
            <option value="american">American</option>
            <option value="british">British</option>
            <option value="cajun">Cajun</option>
            <option value="caribbean">Caribbean</option>
            <option value="chinese">Chinese</option>
            <option value="french">French</option>
            <option value="german">German</option>
            <option value="greek">Greek</option>
            <option value="indian">Indian</option>
            <option value="irish">Irish</option>
            <option value="italian">Italian</option>
            <option value="japanese">Japanese</option>
            <option value="jewish">Jewish</option>
            <option value="korean">Korean</option>
            <option value="latin American">Latin</option>
            <option value="mexican">Mexican</option>
            <option value="middle Eastern">Mediterranean</option>
            <option value="nordic">Nordic</option>
            <option value="spanish">Spanish</option>
            <option value="southern">Southern</option>
            <option value="thai">Thai</option>
            <option value="vietnamese">Vietnamese</option>
          </select>
          </li>
          <li>
          <select>
            <option value="" selected disabled hidden>Allergies</option>
            <option value="dairy">Dairy</option>
            <option value="egg">Egg</option>
            <option value="gluten">Gluten</option>
            <option value="peanut">Peanut</option>
            <option value="sesame">Sesame</option>
            <option value="seafood">Seafood</option>
            <option value="shellfish">Shellfish</option>
            <option value="soy">Soy</option>
            <option value="sulfite">Sulfite</option>
            <option value="tree Nut">Tree Nut</option>
            <option value="wheat">Wheat</option>
          </select>
          </li>
          <li>
          <select>
            <option value="" selected disabled hidden>Type</option>
            <option value="appetizer">Appetizer</option>
            <option value="breakfast">Breakfast</option>
            <option value="beverage">Beverage</option>
            <option value="bread">Bread</option>
            <option value="drink">Cocktail</option>
            <option value="dessert">Dessert</option>
            <option value="main course">Main</option>
            <option value="salad">Salad</option>
            <option value="sauce">Sauce</option>
            <option value="soup">Soup</option>
          </select>
          </li>
          </ul> */}
            <ul className="shopping-list js-shopping-list">
              {ingredientsList.map((ingredient, i) =>
                <li key={i} className="shopping-item">
                  <Ingredient
                    key={i}
                    id={ingredient.stateId}
                    name={ingredient.name}                    />
                </li>
              )}
            </ul>
        </section>
          )
        }
      }
