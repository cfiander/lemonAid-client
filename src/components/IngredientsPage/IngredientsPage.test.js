
import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import IngredientsPage from './IngredientsPage';

describe(`IngredientsPage component`, () => {
  const props = {
    notes: [
      {
        "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Apple",
      },
      {
        "stateId": "d26e0034-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Sugar",
      },
      {
        "stateId": "d26e01a6-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Caramel",
      },
      {
        "stateId": "d26e0570-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Flour",
      },
    ]
  }

  it('renders an IngredientsPage by default', () => {
    const wrapper = shallow(<IngredientsPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a Ingredient in ul for each Ingredient in array', () => {
    const ul = shallow(<IngredientsPage {...props} />)
      .find('ul')
    expect(toJson(ul)).toMatchSnapshot()
  })
})