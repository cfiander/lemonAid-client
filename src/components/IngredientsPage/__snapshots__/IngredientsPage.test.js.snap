// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngredientsPage component renders a Ingredient in ul for each Ingredient in array 1`] = `
Array [
  <ul
    className="filter-controls"
  >
    <li>
      <select>
        <option
          disabled={true}
          hidden={true}
          selected={true}
          value=""
        >
          Diet
        </option>
        <option
          value="keto"
        >
          Keto
        </option>
        <option
          value="aleo"
        >
          Paleo
        </option>
        <option
          value="rimal"
        >
          Primal
        </option>
        <option
          value="frican"
        >
          Pescetarian
        </option>
        <option
          value="egan"
        >
          Vegan
        </option>
        <option
          value="egetarian"
        >
          Vegetarian
        </option>
      </select>
    </li>
    <li>
      <select>
        <option
          disabled={true}
          hidden={true}
          selected={true}
          value=""
        >
          Cuisine
        </option>
        <option
          value="african"
        >
          African
        </option>
        <option
          value="american"
        >
          American
        </option>
        <option
          value="british"
        >
          British
        </option>
        <option
          value="cajun"
        >
          Cajun
        </option>
        <option
          value="caribbean"
        >
          Caribbean
        </option>
        <option
          value="chinese"
        >
          Chinese
        </option>
        <option
          value="french"
        >
          French
        </option>
        <option
          value="german"
        >
          German
        </option>
        <option
          value="greek"
        >
          Greek
        </option>
        <option
          value="indian"
        >
          Indian
        </option>
        <option
          value="irish"
        >
          Irish
        </option>
        <option
          value="italian"
        >
          Italian
        </option>
        <option
          value="japanese"
        >
          Japanese
        </option>
        <option
          value="jewish"
        >
          Jewish
        </option>
        <option
          value="korean"
        >
          Korean
        </option>
        <option
          value="latin American"
        >
          Latin
        </option>
        <option
          value="mexican"
        >
          Mexican
        </option>
        <option
          value="middle Eastern"
        >
          Mediterranean
        </option>
        <option
          value="nordic"
        >
          Nordic
        </option>
        <option
          value="spanish"
        >
          Spanish
        </option>
        <option
          value="southern"
        >
          Southern
        </option>
        <option
          value="thai"
        >
          Thai
        </option>
        <option
          value="vietnamese"
        >
          Vietnamese
        </option>
      </select>
    </li>
    <li>
      <select>
        <option
          disabled={true}
          hidden={true}
          selected={true}
          value=""
        >
          Allergies
        </option>
        <option
          value="dairy"
        >
          Dairy
        </option>
        <option
          value="egg"
        >
          Egg
        </option>
        <option
          value="gluten"
        >
          Gluten
        </option>
        <option
          value="peanut"
        >
          Peanut
        </option>
        <option
          value="sesame"
        >
          Sesame
        </option>
        <option
          value="seafood"
        >
          Seafood
        </option>
        <option
          value="shellfish"
        >
          Shellfish
        </option>
        <option
          value="soy"
        >
          Soy
        </option>
        <option
          value="sulfite"
        >
          Sulfite
        </option>
        <option
          value="tree Nut"
        >
          Tree Nut
        </option>
        <option
          value="wheat"
        >
          Wheat
        </option>
      </select>
    </li>
    <li>
      <select>
        <option
          disabled={true}
          hidden={true}
          selected={true}
          value=""
        >
          Type
        </option>
        <option
          value="appetizer"
        >
          Appetizer
        </option>
        <option
          value="breakfast"
        >
          Breakfast
        </option>
        <option
          value="beverage"
        >
          Beverage
        </option>
        <option
          value="bread"
        >
          Bread
        </option>
        <option
          value="drink"
        >
          Cocktail
        </option>
        <option
          value="dessert"
        >
          Dessert
        </option>
        <option
          value="main course"
        >
          Main
        </option>
        <option
          value="salad"
        >
          Salad
        </option>
        <option
          value="sauce"
        >
          Sauce
        </option>
        <option
          value="soup"
        >
          Soup
        </option>
      </select>
    </li>
  </ul>,
  <ul
    className="shopping-list js-shopping-list"
  />,
]
`;

exports[`IngredientsPage component renders an IngredientsPage by default 1`] = `
<section>
  <div
    className="searchForm"
  >
    <Search
      addIngredient={[Function]}
    />
  </div>
  <ul
    className="filter-controls"
  >
    <li>
      <select>
        <option
          disabled={true}
          hidden={true}
          selected={true}
          value=""
        >
          Diet
        </option>
        <option
          value="keto"
        >
          Keto
        </option>
        <option
          value="aleo"
        >
          Paleo
        </option>
        <option
          value="rimal"
        >
          Primal
        </option>
        <option
          value="frican"
        >
          Pescetarian
        </option>
        <option
          value="egan"
        >
          Vegan
        </option>
        <option
          value="egetarian"
        >
          Vegetarian
        </option>
      </select>
    </li>
    <li>
      <select>
        <option
          disabled={true}
          hidden={true}
          selected={true}
          value=""
        >
          Cuisine
        </option>
        <option
          value="african"
        >
          African
        </option>
        <option
          value="american"
        >
          American
        </option>
        <option
          value="british"
        >
          British
        </option>
        <option
          value="cajun"
        >
          Cajun
        </option>
        <option
          value="caribbean"
        >
          Caribbean
        </option>
        <option
          value="chinese"
        >
          Chinese
        </option>
        <option
          value="french"
        >
          French
        </option>
        <option
          value="german"
        >
          German
        </option>
        <option
          value="greek"
        >
          Greek
        </option>
        <option
          value="indian"
        >
          Indian
        </option>
        <option
          value="irish"
        >
          Irish
        </option>
        <option
          value="italian"
        >
          Italian
        </option>
        <option
          value="japanese"
        >
          Japanese
        </option>
        <option
          value="jewish"
        >
          Jewish
        </option>
        <option
          value="korean"
        >
          Korean
        </option>
        <option
          value="latin American"
        >
          Latin
        </option>
        <option
          value="mexican"
        >
          Mexican
        </option>
        <option
          value="middle Eastern"
        >
          Mediterranean
        </option>
        <option
          value="nordic"
        >
          Nordic
        </option>
        <option
          value="spanish"
        >
          Spanish
        </option>
        <option
          value="southern"
        >
          Southern
        </option>
        <option
          value="thai"
        >
          Thai
        </option>
        <option
          value="vietnamese"
        >
          Vietnamese
        </option>
      </select>
    </li>
    <li>
      <select>
        <option
          disabled={true}
          hidden={true}
          selected={true}
          value=""
        >
          Allergies
        </option>
        <option
          value="dairy"
        >
          Dairy
        </option>
        <option
          value="egg"
        >
          Egg
        </option>
        <option
          value="gluten"
        >
          Gluten
        </option>
        <option
          value="peanut"
        >
          Peanut
        </option>
        <option
          value="sesame"
        >
          Sesame
        </option>
        <option
          value="seafood"
        >
          Seafood
        </option>
        <option
          value="shellfish"
        >
          Shellfish
        </option>
        <option
          value="soy"
        >
          Soy
        </option>
        <option
          value="sulfite"
        >
          Sulfite
        </option>
        <option
          value="tree Nut"
        >
          Tree Nut
        </option>
        <option
          value="wheat"
        >
          Wheat
        </option>
      </select>
    </li>
    <li>
      <select>
        <option
          disabled={true}
          hidden={true}
          selected={true}
          value=""
        >
          Type
        </option>
        <option
          value="appetizer"
        >
          Appetizer
        </option>
        <option
          value="breakfast"
        >
          Breakfast
        </option>
        <option
          value="beverage"
        >
          Beverage
        </option>
        <option
          value="bread"
        >
          Bread
        </option>
        <option
          value="drink"
        >
          Cocktail
        </option>
        <option
          value="dessert"
        >
          Dessert
        </option>
        <option
          value="main course"
        >
          Main
        </option>
        <option
          value="salad"
        >
          Salad
        </option>
        <option
          value="sauce"
        >
          Sauce
        </option>
        <option
          value="soup"
        >
          Soup
        </option>
      </select>
    </li>
  </ul>
  <ul
    className="shopping-list js-shopping-list"
  />
</section>
`;
